src/ppom-controller.ts

```diff
@@ -273,16 +278,26 @@ export class PPOMController extends BaseControllerV2<
       });
     });
 
-    this.#registerMessageHandlers();
-    this.#startDataRefreshTask();
-  }
+    onPreferencesChange((preferenceControllerState: any) => {
```

Comment:
+1, I will add disable, enable functions.

I may still still need controller instance to be created each time so that it can be injected to JSON RPC engine at initialisation. I need to check this more.

Reply:
Additionally, it does not appear to be great idea to put blockaid specific code in MetamaskController. It is huge controller due to a lot of such code.